pkg/file/deferred_partial_read_closer.go

```diff
@@ -0,0 +1,73 @@
+package file
+
+import (
+	"errors"
+	"io"
+	"os"
+)
+
+var _ io.ReadCloser = (*deferredPartialReadCloser)(nil)
+
+// deferredPartialReadCloser is a "lazy" read closer, allocating a file descriptor for the given path only upon the first Read() call.
+// Additionally only part of the file is allowed to be read, starting at a given position.
+type deferredPartialReadCloser struct {
```

Comment:
nit / for consideration: The name `deferredPartialReadCloser` is not immediately obvious, but I see the motivation, especially for the descriptors "deferred" and "partial". It might be good to consider other descriptors in an attempt to increases the obviousness. For example, once I saw "lazy" in your comment, I immediately grasped that concept. Maybe "lazy" in place of "deferred"? (Also "defer" has a particular connotation in Go.) For "partial", my impression based on the implementation is that it's not necessarily partial — the consumer can specify any starting point and size, including values such that it's actually the whole file. I'm thinking "bounded" or something might be closer? What do you think?

Reply:
Once I understood the concept, though, I loved it. This "object" is a very useful construct. ❤️

Reply:
👍  went with "LazyBoundedReadCloser" and also renamed it's sibling "LazyReadCloser"